### ISSUE TITLE
Wrong mapping for `portuguese` language

### ISSUE THREAD

**Issue by rgcv** [valentjn/languagetool-languageserver#72]
Hey there!

First off, great extension, I love it, awesome stuff with L<sup>A</sup>T<sub>E</sub>X support.

I've seemed to have identified what looks to be an issue when trying to enclose portuguese text using babel commands.  I've tried them all: `foreignlanguage`, `selectlanguage`, the `otherlanguage` environment.  Magic comments work fine with the language tag, but that had me thinking.

I resorted, as documented, to use the `babeltags` command in my document's preamble like so

```tex
\babeltags{pt=portuguese}
```

To my surprise, and confusion, when using `textpt` or the now available `pt` environment, checking worked wonderfully.

I tracked down when support for babel commands were introduced and noticed this line over here when initializing the `babelLanguageMap`

https://github.com/valentjn/ltex-ls/blob/5832d4dedef6e952aef8873e29eed5bf6ccd2d19/ltexls/src/main/java/org/bsplines/ltexls/parsing/latex/LatexFragmentizer.java#L157

The language `portugese` is not made available by babel, only making available the codenames `portuguese` (spelled correctly) and the historically short `portuges`.  I'm not sure this is either a typo or that babel used to support `portugese` as well in the past.  If it was a typo, I suggest its removal, keeping it otherwise.

Regardless, I would recommend adding the proper mappings.  While I'm at it, I'd also suggest adding the historical aliases for compatibility reasons, if those are interesting to support as well.  Some java pseudo-code follows:

```java
babelLanguageMap.put("brazil", "pt-BR");
babelLanguageMap.put("brazilian", "pt-BR");
babelLanguageMap.put("portuges", "pt");
babelLanguageMap.put("portuguese", "pt");
```

Quick disclaimer: I'm not sure this is enough to fix the whole problem, but from what little I've looked at, it seems to be enough.  If it is, I can gladly submit a PR adding the missing mappings.  Removing the `portugese` mapping is up for debate.  I honestly don't recall if that language codename was supported once, but it doesn't appear to be documented in the most recent version of the documentation.

**Comment by valentjn** [comment]
Thanks for the report. I copied the languages from the babel manual by hand, so that was a typo. Also, I think `pt-PT` should be used instead of `pt`. Not using the regional variant, if one exists, results in specific errors (mostly spelling errors) not being displayed, at least for other languages.

**Comment by rgcv** [comment]
No problem, thank you for the quick solution!  I'm a typist myself, I understand it perfectly.  And I agree, if one exists, the regional variant should be used, sounds sensible.

**Comment by valentjn** [comment]
Fix released in 12.3.0.
